src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1979,8 +1979,9 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 	}
 	/* file/dir should be handled by DFS */
 	if (oflags & O_CREAT) {
-		rc = dfs_open(dfs_mt->dfs, parent, item_name, mode | S_IFREG, oflags & (~O_APPEND),
-			      0, 0, NULL, &dfs_obj);
+		/* clear the bits for types first */
+		rc = dfs_open(dfs_mt->dfs, parent, item_name, (mode & (~S_IFMT)) | S_IFREG,
+			      oflags & (~O_APPEND), 0, 0, NULL, &dfs_obj);
```

Comment:
I am confused by this in the first place.. what if we are creating a directory? i.e. why do we have | S_IFREG ?

Reply:
I thought mkdir is normal used for creating dir. I assumed open is not used to create a directory.
I will add the code to handle "O_DIRECTORY".

Reply:
im not familiar of when we end up here.. im just referring to your comment on line 1980

Reply:
to be clear if this is just for files, that's fine, but your comment is wrong.
I also assumed open_common() is for both files and dirs, but his is your code, so you should clarify

Reply:
I will run some tests with O_DIRCTORY and O_PATH combined with O_CREAT to check the behavior of glibc on regular file system. Then decide how to proceed about this issue.